test/integration/instance.validations.test.ts

```diff
@@ -0,0 +1,978 @@
+            declare id: number | string;
+          }
+
+          User.init({
+            id: {
+              type: DataTypes.INTEGER,
+              autoIncrement: true,
+              primaryKey: true,
+              validate: {
+                isInt: {
+                  msg: 'ID must be an integer!',
+                },
+              },
+            },
+          }, { sequelize });
+
+          await User.sync({ force: true });
+
+          return { User };
+        });
+
+        it('should emit an error when we try to enter in a string for the id key with validation arguments', async () => {
+          try {
+            await vars.User.create({ id: 'helloworld' });
+          } catch (error) {
+            expect(error).to.be.an.instanceOf(Error);
+            if (error instanceof ValidationError) {
+              expect(error.get('id')[0].message).to.equal('ID must be an integer!');
+            }
+          }
+        });
+
+        it('should emit an error when we try to enter in a string for an auto increment key through .build().validate()', async () => {
+          const user = vars.User.build({ id: 'helloworld' });
+
+          const err = await expect(user.validate()).to.be.rejected;
+          expect(err.get('id')[0].message).to.equal('ID must be an integer!');
+        });
+
+        it('should emit an error when we try to .save()', async () => {
+          const user = vars.User.build({ id: 'helloworld' });
+
+          try {
+            await user.save();
+          } catch (error) {
+            expect(error).to.be.an.instanceOf(Error);
+            if (error instanceof ValidationError) {
+              expect(error.get('id')[0].message).to.equal('ID must be an integer!');
+            }
+          }
+        });
+      });
+    });
+
+    describe('pass all paths when validating', async () => {
+      const vars = beforeEach2(async () => {
+        class Project extends Model<InferAttributes<Project>, InferCreationAttributes<Project>> {
+          static associations: {
+            users: HasMany<Project, Task>,
+          };
+
+          declare name: CreationOptional<string>;
+        }
+
+        Project.init({
+          name: {
+            type: DataTypes.STRING,
+            allowNull: false,
+            defaultValue: 'unknown',
+            validate: {
+              isIn: [['unknown', 'hello', 'test']],
+            },
+          },
+        }, { sequelize });
+
+        class Task extends Model<InferAttributes<Task>, InferCreationAttributes<Task>> {
+          static associations: {
+            users: BelongsTo<Task, Project>,
+          };
+
+          declare something: number;
+        }
+
+        Task.init({
+          something: DataTypes.INTEGER,
+        }, { sequelize });
+
+        Project.hasOne(Task);
+        Task.belongsTo(Project);
+
+        await Project.sync({ force: true });
+        await Task.sync({ force: true });
+
+        return { Project, Task };
+      });
```

Comment:
this beforeEach is only used in a single test. This whole `describe` should be a single `it` instead